chpl/chpl-service/src/main/java/gov/healthit/chpl/listener/QuestionableActivityListener.java

```diff
@@ -45,40 +45,39 @@ public void setEnvironment(final Environment e) {
-            Object originalData, Object newData) {
-        checkQuestionableActivity(concept, objectId, activityDescription, originalData, newData, null);
+            Date activityDate, Object originalData, Object newData) {
+        checkQuestionableActivity(concept, objectId, activityDescription, activityDate, originalData, newData, null);
     }
 
     public void checkQuestionableActivity(ActivityConcept concept, Long objectId, String activityDescription,
-            Object originalData, Object newData, String reason) {
+            Date activityDate, Object originalData, Object newData, String reason) {
 
         if (newData instanceof CertifiedProductSearchDetails) {
-            checkQuestionableActivityForListingEdit(concept, objectId, activityDescription,
+            checkQuestionableActivityForListingEdit(concept, objectId, activityDescription, activityDate,
                     (CertifiedProductSearchDetails) originalData, (CertifiedProductSearchDetails) newData, reason);
         } else if (areAllValuesNonNull(originalData, newData) && originalData instanceof Developer
                 && newData instanceof Developer) {
-            checkQuestionableActivityForDeveloper(concept, objectId, activityDescription, (Developer) originalData,
+            checkQuestionableActivityForDeveloper(concept, objectId, activityDescription, activityDate, (Developer) originalData,
                     (Developer) newData);
         } else if (areAllValuesNonNull(originalData, newData) && originalData instanceof Product
                 && newData instanceof Product) {
-            checkQuestionableActivityForProduct(concept, objectId, activityDescription, (Product) originalData,
+            checkQuestionableActivityForProduct(concept, objectId, activityDescription, activityDate, (Product) originalData,
                     (Product) newData);
         } else if (areAllValuesNonNull(originalData, newData) && originalData instanceof ProductVersionDTO
                 && newData instanceof ProductVersionDTO) {
-            checkQuestionableActivityForVersion(concept, objectId, activityDescription, (ProductVersionDTO) originalData,
+            checkQuestionableActivityForVersion(concept, objectId, activityDescription, activityDate, (ProductVersionDTO) originalData,
                     (ProductVersionDTO) newData);
         }
     }
 
     private void checkQuestionableActivityForListingEdit(ActivityConcept concept, Long objectId, String activityDescription,
-            CertifiedProductSearchDetails originalData, CertifiedProductSearchDetails newData, String reason) {
+            Date activityDate, CertifiedProductSearchDetails originalData, CertifiedProductSearchDetails newData, String reason) {
 
         if (originalData == null || newData == null || AuthUtil.getCurrentUser() == null) {
             return;
         }
 
         // all questionable activity from this action should have the exact same date and user id
```

Comment:
There are a few comments in this file that don't quite make sense if the date is now a passed in value. I know comments aren't super important, but _wrong_ comments seem like something we shouldn't allow